content/source/docs/cloud/workspaces/agent.html.md

```diff
@@ -0,0 +1,175 @@
+
+![Screenshot: The Agents page with one connected agent in the idle state](./images/agent-running.png)
+
+##### Optional Configuration: Running an Agent using Docker
+
+Alternatively, you can use our official agent Docker container to run the Agent.
+
+```
+docker pull hashicorp/tfc-agent:latest
+docker run -e TFC_AGENT_TOKEN=your-token -e
+TFC_AGENT_NAME=your-agent-name hashicorp/tfc-agent
+```
+
+##### Optional Configuration: Single-execution mode
+
+The Agent can also be configured to run in single-execution mode, which will ensure that the Agent only runs a single workload, then terminates. This can be used in combination with Docker and a process supervisor to ensure a clean working environment for every Terraform run.
+
+To use single-execution mode, start the agent with the `-single` command line argument.
+
+### Configuring Workspaces to use the Agent
+
+-> **Note:** You must have “Admin” access to the workspace you are configuring to change the [execution mode](/docs/cloud/workspaces/settings.html#execution-mode). ([More about permissions.](/docs/cloud/users-teams-organizations/permissions.html))
+
+[permissions-citation]: #intentionally-unused---keep-for-maintainers
+
+~> **Important:** Changing your workspace's [execution mode](/docs/cloud/workspaces/settings.html#execution-mode) after a run has already been planned will cause the run to error when it is applied.
+To minimize the number runs that error, you should disable [auto-apply](/docs/cloud/workspaces/settings.html#auto-apply-and-manual-apply),
+complete any runs that are no longer in the [pending stage](/docs/cloud/run/states.html#1-the-pending-stage),
+and [lock](/docs/cloud/workspaces/settings.html#locking) your workspace before changing the execution mode.
+
+To configure a workspace to execute runs using an agent:
+
+1. Open the workspace from the main Workspaces view, then navigate to Settings > General from the dropdown menu.
+1. Select Agent as the [execution mode](/docs/cloud/workspaces/settings.html#execution-mode).
+1. Click Save Settings at the bottom of the page.
+
+![Screenshot: The Workspace General settings page, with agent selected for execution mode](./images/agent-workspace-execution-mode.png)
+
+#### Running Multiple Agents
+
+You may choose to run multiple agents within your network for improved resiliency. If there are multiple agents available within an organization, Terraform Cloud will select the first available agent.
+
+#### Revoking an Agent Token
+
+You may revoke an issued token from your agents at any time.
+
+Revoking a token will cause the agents using it to become unable to process work until they are reinitialized with a new token. Under normal circumstances, it may be desirable to generate a new token first, initialize the agents using it, then revoke the old token once no agents are using it. Agent tokens display information about the last time they were used to help you decide whether they are safe to revoke.
```

Comment:
so as I discovered yesterday, the agents actually exit if the token is revoked. Should update this to reflect that but I'm not 100% certain what the behavior is on the binary version vs the container (the container exits, does the binary as well?)

Reply:
the container exits _because_ the binary exits :) Yes, this is consistent between the two.

Reply:
thanks! so, how about this then

```suggestion
Revoking a token will cause the agents using it exit. For the agent pool to continue to process work, member agents must be reinitialized with a new token. Under normal circumstances, it may be desirable to generate a new token first, initialize the agents using it, then revoke the old token once no agents are using it. Agent tokens display information about the last time they were used to help you decide whether they are safe to revoke.
```

Reply:
👍